sql/sys_vars.h

```diff
@@ -831,8 +851,10 @@ class Sys_var_charptr : public sys_var {
     var->save_result.string_value.length = ptr ? strlen(ptr) : 0;
   }
   void saved_value_to_string(THD *, set_var *var, char *def_val) {
-    memcpy(def_val, var->save_result.string_value.str,
-           var->save_result.string_value.length);
+    if (var->save_result.string_value.str) {
```

Comment:
please change it != nullptr
and in which case this would be called with nullptr?

Reply:
In case of empty string.